webextension/privileged/ui/CloudDownloadsView.jsm

```diff
@@ -545,19 +584,42 @@ var CloudDownloadsView = {
         // Prepare for future downloads move to Download Folder by checking
         // if Download folder exists in cloud provider folder, if not create one
         CloudDownloadsInternal.checkProviderDownloadFolder();
+
+        // Once opted-in, set observers to send subsequent download pref changes to telemetry
+        Services.prefs.addObserver("browser.download.folderList", CloudDownloadsInternal.downloadPrefObserve);
+        Services.prefs.addObserver("browser.download.useDownloadDir", CloudDownloadsInternal.downloadPrefObserve);
```

Comment:
You never remove these preference observers.

It's probably easier to just piggyback on the existing lazy preference getter we have for this, and check in the pref update callback whether the user has opted in (ie whether we've saved a prompt response). You'd need to add another lazy pref getter for the other pref though.

Reply:
Done